agents/mysql/slowlog/slowlog.go

```diff
@@ -156,6 +176,36 @@ func (s *SlowLog) Run(ctx context.Context) {
 	}
 }
 
+func (s *SlowLog) rotateSlowLog(ctx context.Context, slowLogPath string) error {
+	db, err := sql.Open("mysql", s.dsn)
+	if err != nil {
+		return errors.Wrap(err, "cannot open database connection")
+	}
+	defer db.Close() //nolint:errcheck
+
+	_, err = db.ExecContext(ctx, fmt.Sprintf("SET GLOBAL /* %s */ slow_query_log=off", queryTag)) //nolint:gosec
+	if err != nil {
+		return errors.Wrap(err, "cannot dissable slow_query_log")
+	}
+
+	_, err = db.ExecContext(ctx, fmt.Sprintf("FLUSH LOGS /* %s */", queryTag)) //nolint:gosec
+	if err != nil {
+		return errors.Wrap(err, "cannot flush logs")
+	}
+
+	err = os.Rename(slowLogPath, slowLogPath+".old")
```

Comment:
We should remove it, not rename. `ContinuousFileReader` will still be able to read removed file.

Reply:
We should also remove it before `FLUSH LOGS`. That the whole point of that SQL command.